Purchases/Purchasing/StoreKit2/ProductsFetcherSK2.swift

```diff
@@ -24,26 +24,65 @@ actor ProductsFetcherSK2 {
     }
 
     private var cachedProductsByIdentifier: [String: SK2StoreProduct] = [:]
+    private var cachedProductsStorefrontIdentifier: String?
 
     func products(identifiers: Set<String>) async throws -> Set<SK2StoreProduct> {
         do {
-            // todo: also cache requests, so that if a request is in flight for the same products,
-            // we don't need to make a new one
-            let productsAlreadyCached = self.cachedProductsByIdentifier.filter { key, _ in identifiers.contains(key) }
-            if productsAlreadyCached.count == identifiers.count {
-                let productsAlreadyCachedSet = Set(productsAlreadyCached.values)
-                Logger.debug(Strings.offering.products_already_cached(identifiers: identifiers))
-                return productsAlreadyCachedSet
+            if let cachedProducts = await self.cachedProducts(withIdentifiers: identifiers) {
+                return cachedProducts
             }
 
             let storeKitProducts = try await StoreKit.Product.products(for: identifiers)
-            let sk2StoreProduct = storeKitProducts.map { SK2StoreProduct(sk2Product: $0) }
-            return Set(sk2StoreProduct)
+            let sk2StoreProducts = Set(storeKitProducts.map { SK2StoreProduct(sk2Product: $0) })
+
+            await self.cache(products: sk2StoreProducts)
+
+            return sk2StoreProducts
         } catch {
             throw Error.productsRequestError(innerError: error)
         }
     }
 
+    private func cachedProducts(withIdentifiers identifiers: Set<String>) async -> Set<SK2StoreProduct>? {
```

Comment:
nitpick: we should move these to a [`private` extension](https://github.com/RevenueCat/purchases-ios/blob/main/SwiftStyleGuide.swift#L124), so that at a first glance it's easier to tell what the non-private part of the class is

Reply:
Yes! So much better. Now only `products(identifiers:)` remains in the main type definition.